### PR TITLE
Use standart functions for ostream conversions

### PR DESCRIPTION
**Pull request by hoip**

### CODE REVIEW

File: lib/cpp/export/ecl/ostream.hpp [hunk 1]
```diff
@@ -2,190 +2,170 @@
...
-        val2 /= 10;
+ #ifndef ECL_OSTREAM_HPP
+ #define ECL_OSTREAM_HPP
+ 
+ #include <stdio.h>
+ #include <ecl/assert.h>
+ namespace ecl 
+ {
+    
+    template< typename stream >
```
Comment (forGGe) [inline lib/cpp/export/ecl/ostream.hpp]: Move `endl` where it was - 4 spaces left.

File: lib/cpp/export/ecl/ostream.hpp [hunk 2]
```diff
@@ -2,190 +2,170 @@
...
-    char out_character;
-
-    while (val2 / 10 != 0) {
-        higher_multiplicand *= 10;
-        val2 /= 10;
+ #ifndef ECL_OSTREAM_HPP
+ #define ECL_OSTREAM_HPP
+ 
+ #include <stdio.h>
+ #include <ecl/assert.h>
```
Comment (forGGe) [inline lib/cpp/export/ecl/ostream.hpp]: Add space here.

File: lib/cpp/export/ecl/ostream.hpp [hunk 3]
```diff
@@ -2,190 +2,170 @@
...
-{
-    unsigned int higher_multiplicand = 1;
-    unsigned int out_digit = 0;
-    int val2 = value;
-    char out_character;
-
-    while (val2 / 10 != 0) {
-        higher_multiplicand *= 10;
-        val2 /= 10;
+ #ifndef ECL_OSTREAM_HPP
```
Comment (forGGe) [inline lib/cpp/export/ecl/ostream.hpp]: Let's preserve an indentation. Remove 1 space indent from whole file

File: lib/cpp/export/ecl/ostream.hpp [hunk 4]
```diff
@@ -2,190 +2,170 @@
...
+ }
+ 
+ 
+ template< class IO_device >
+ ostream<IO_device> &ostream< IO_device >::operator<<(char character)
+ {
+     
+    char buff [32];
+    
+    char n = snprintf(buff, sizeof(buff), "%c", character);
```
Comment (forGGe) [inline lib/cpp/export/ecl/ostream.hpp]: No need to write character in buffer and then put it in the device. Let's just call `put()` here.

File: lib/cpp/export/ecl/ostream.hpp [hunk 5]
```diff
@@ -2,190 +2,170 @@
...
+ 
+ template< class IO_device >
+ ostream<IO_device> &ostream< IO_device >::operator<<(const char *string)
+ {
+     // TODO: improve this function in a way it can write a string
+     // to a device with chunks that are split by '\n' symbol
+     // What about somehow move it to the platform layer?
+ 
+     size_t i = 0;
+     while (string[i] != '\0')
```
Comment (forGGe) [inline lib/cpp/export/ecl/ostream.hpp]: According to the code style - place brackets on 1st line of `while` and similar statements.
See Rule 1. in https://forgge.github.io/theCore/community.html#bracket-placement

File: lib/cpp/export/ecl/ostream.hpp [hunk 6]
```diff
@@ -2,190 +2,170 @@
...
+     {
+         if (string[i] == '\n') 
+         {
+             uint8_t carret_ret = '\r';
+             m_device->write(&carret_ret, 1);
+         }
+ 
+         if (m_device->write((const uint8_t *) &string[i], 1) < 0) 
+         {
+             //Serial.printlnf("\n>%d<\n", count);
```
Comment (forGGe) [inline lib/cpp/export/ecl/ostream.hpp]: This line can be deleted.
See Rule 1. https://forgge.github.io/theCore/community.html#comments

File: lib/cpp/export/ecl/ostream.hpp [hunk 7]
```diff
@@ -2,190 +2,170 @@
...
+ ostream<IO_device> &ostream< IO_device >::operator<<(
+         ostream& (*func)(ostream< IO_device >&))
+ {
+     return func(*this);
+ }
+ 
+ template< class IO_device >
+ ostream<IO_device> &ostream< IO_device >::put(char c)
+ {  
+     char buff [32];
```
Comment (forGGe) [inline lib/cpp/export/ecl/ostream.hpp]: No need to write character in buffer and then put it in the device. Previous version of this function was good, let's bring it back.

**Comment by forGGe** [comment]
Related to #204

**Review by forGGe — CHANGES_REQUESTED** [review]
Looks good, only minor nitpicks.

Also, please squash all commits together with some meaningful message, like
`>204 Use standart functions for ostream conversions`

See also:
https://forgge.github.io/theCore/community.html#commit-and-pr-style , Rule 3.

**Review by forGGe — APPROVED** [review]

**Comment by forGGe** [comment]
Changes are approved, but hoip/theCore#1 should be merged first.

**Comment by forGGe** [comment]
Merged. Thanks @hoip !

**Comment by hoip** [comment]
Please!

**Pull request merged**